### PR TITLE
imp: remove the channel type = ordered checks from both host and controller

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #5522 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by srdtrk — APPROVED** [review]
Should these ICA PRs go on a feature branch? I'm asking since allowing this without merging the rest of the ICA PRs seems a bit dangerous

**Comment by DimitrisJim** [comment]
> Should these ICA PRs go on a feature branch?

Yea, they are https://github.com/cosmos/ibc-go/tree/feat/allow-unordered-ica-channels @charleenfei can just switch base branches

**Comment by charleenfei** [comment]
> Should these ICA PRs go on a feature branch? I'm asking since allowing this without merging the rest of the ICA PRs seems a bit dangerous

makes sense! i can open it against a feature branch, i just saw cian created one already, didn't see that before!

File: modules/apps/27-interchain-accounts/controller/keeper/handshake.go [hunk 1]
```diff
@@ -31,10 +31,6 @@ func (k Keeper) OnChanOpenInit(
 	counterparty channeltypes.Counterparty,
 	version string,
 ) (string, error) {
-	if order != channeltypes.ORDERED {
```
Comment (DimitrisJim) [inline modules/apps/27-interchain-accounts/controller/keeper/handshake.go:34]: docustring can be updated ahora to remove mention of 'must be ORDERED'.

**Review by DimitrisJim — APPROVED** [review]
lgtm, one nit w.r.t docu

**Pull request merged**